app/controllers/api/api_controller.rb

```diff
@@ -101,6 +97,120 @@ def dashboard
     end
   end
 
+  # Get Team & Team Role & Permissions info for the Mobile App.
+  def mobile_basic
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    user_team = Team.find_by_id(current_user.team_id)
+    user_team_role = TeamRole.find_by_id(current_user.team_role_id)
+
+    begin
+      #generate overall embedded result
+      @result = {
+        'team' => user_team,
+        'team_role' => user_team_role
+      }
+    rescue
+      @status = 500
+      @message = "Failure to generate mobile basic results."
+    end
+
+
+    # if we haven't set a message, we sucessfully made stuff
+    unless @message
+      @status = 200
+      @message = "Success!"
+    end
+
+    @response = {
+      status: @status,
+      message: @message,
+      date:  Time.now,
+      result: @result
+    }
+
+    respond_to do |format|
+      format.json { render :json => @response }
+    end
+  end
+
+  # Get Public Relations / Income info for the Mobile App - Income Tab.
+  def income
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    round = @game.round
+    public_relations_list = PublicRelation.where(game: @game)
+
+    begin
+      #generate overall embedded result
+      @result = {
+        "timer" => {
+          "round"=>  round,
+        },
+        "pr" => public_relations_list
+      }
+    rescue
+      @status = 500
+      @message = "Failure to generate public relation results."
+    end
+
+    # if we haven't set a message, we sucessfully made stuff
+    unless @message
+      @status = 200
+      @message = "Success!"
+    end
+
+    @response = {
+      status: @status,
+      message: @message,
+      date:  Time.now,
+      result: @result
+    }
+
+    respond_to do |format|
+      format.json { render :json => @response }
+    end
+  end
+
+ # Get Message Data for the Mobile App - Messages Tab.
+  def messages
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    round = @game.round
+    messages = Message.all.order('created_at DESC')
```

Comment:
@FifthSurprise 

I have the changes we spoke about mostly done, but my attempts at retrieving Message information are blocked by my lack of Ruby knowledge. What I wrote so far is:

```
receivedMessages = Message.where(game: @game).where(recipient_id: user_team_id).group_by{|x| x[:sender_id]}
sentMessages = Message.where(game: @game).where(sender_id: user_team_id).group_by{|x| x[:recipient_id]}
```

This retrieves all the messages that we have either sent or received and groups them by who the conversation is with. This works separately, but what I'd like to do is alter this to actually give us more data in a more complex structure. 

I want to treat the messages themselves essentially as an array of `message` objects as one property of a larger `conversation` object. The object would also contain properties for the Team Name and an "updated_at" property that is equal to the latest message's timestamp.

Essentially the final data structure that I've imagined is an array of conversation objects, that look more or less like this:

```
{
  conversation_partner: "Japan",
  updated_at: '2016-03-20T16:15:00.000Z',
  messages: [
    {
      sender_id: 4,
      recipient_id: 1,
       content: "I think China is conspiring against you.",
       created_at: '2016-03-20T10:30:00.000Z'
     },
     {
      sender_id: 1,
      recipient_id: 4,
      content: "What have you heard?",
      created_at: '2016-03-20T16:15:00.000Z'
     }
   ]
},
```

Whereas what I have right now is this:

```
sentMessages = [
  {
    sender_id: 4,
    recipient_id: 1,
     content: "I think China is conspiring against you.",
     created_at: '2016-03-20T10:30:00.000Z'
  }
]

receivedMessages = [
  {
    sender_id: 1,
    recipient_id: 4,
    content: "What have you heard?",
    created_at: '2016-03-20T16:15:00.000Z'
  }
]
```

Let me know if that doesn't actually make sense.